File: RockitCompiler/benchmarks/bench_list_build.cpp
#include <cstdio>
#include <cstdint>
#include <vector>

int main() {
    int64_t n = 500000;
    std::vector<int64_t> arr;
    for (int64_t i = 0; i < n; i++) {
        arr.push_back(i);
    }
    int64_t sum = 0;
    for (int64_t v : arr) {
        sum += v;
    }
    printf("%lld\n", sum);
    return 0;
}

File: RockitCompiler/benchmarks/bench_list_build.rs
fn main() {
    let n: i64 = 500000;
    let mut arr: Vec<i64> = Vec::new();
    for i in 0..n {
        arr.push(i);
    }
    let sum: i64 = arr.iter().sum();
    println!("{}", sum);
}
